fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/client/fluid/FluidVariantRenderHandler.java

```diff
@@ -76,14 +78,29 @@ default Sprite getSprite(FluidVariant fluidVariant) {
 	}
 
 	/**
-	 * Return the color to use when rendering {@linkplain #getSprite the sprite} of this fluid variant.
+	 * @deprecated Use and implement {@linkplain #getColor(FluidVariant, BlockRenderView, BlockPos) the other more general overload}.
+	 * This one will be removed in a future iteration of the API.
 	 */
+	@Deprecated(forRemoval = true)
 	default int getColor(FluidVariant fluidVariant) {
+		return getColor(fluidVariant, null, null);
+	}
+
+	/**
+	 * Return the color to use when rendering {@linkplain #getSprite the sprite} of this fluid variant.
+	 * Transparency (alpha) will generally be taken into account and should be specified as well.
+	 *
+	 * <p>The world and position are optional context parameters and may be {@code null}.
+	 * If they are null, this method must return a location-independent color.
+	 * If they are provided, this method may return a color that depends on the location.
+	 * For example, water returns the biome-dependent color if the context parameters are specified, or its default color if one of them is null.
+	 */
+	default int getColor(FluidVariant fluidVariant, @Nullable BlockRenderView view, @Nullable BlockPos pos) {
 		// Use the fluid render handler by default.
 		FluidRenderHandler fluidRenderHandler = FluidRenderHandlerRegistry.INSTANCE.get(fluidVariant.getFluid());
 
 		if (fluidRenderHandler != null) {
-			return fluidRenderHandler.getFluidColor(null, null, fluidVariant.getFluid().getDefaultState());
+			return fluidRenderHandler.getFluidColor(view, pos, fluidVariant.getFluid().getDefaultState()) | 255 << 24;
```

Comment:
Hmm, should there be an option to pass in the fluid state to the original handler as well?

Reply:
I don't think a `FluidVariant` can be mapped to a `FluidState` easily. If the fluid state depends on the NBT of the variant, then I'd suggest that a custom `FluidVariantRenderHandler` be registered by the fluid.